New Extension "TT": Translator

deptno opened the pull request:
## Description

Add `TT` extension to translate texts with multiple providers.  
- `From` language, `To` language are configurable.
- Multi language for `TT` extension will be supported. 

## Screencast
https://user-images.githubusercontent.com/1223020/156202118-7c50308d-89ed-49db-8b4b-f23ba62cfd2d.mov

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed in the `media` folder

pernielsentikaer reviewed (commented):
Hi @deptno 👋 

Thanks for your contribution 🔥, well done! I have some comments for you:

- There is already an open PR with papago search #888 started by @wooogle – it would make sense to use one of the extension as a base and both contribute.

### Some feedback for your extension
- At the moment, Raycast doesn't support localization and only supports US English.
- The name of the command needs to be understandable, check [Naming convention](https://developers.raycast.com/basics/prepare-an-extension-for-store#extensions-and-commands-naming)
- Do I need to register for a key before “history” and does that make any sense?
- We do already have [Google Translate](https://www.raycast.com/ospfranco/translate) so it makes sense to make the Papago translater and focus on that?

Let me know your thoughts

deptno commented:
Hi @pernielsentikaer

Thank you for your feedback. I reflected them in this extension.

- At the moment, Raycast doesn't support localization and only supports US English.  
  - I removed code related i18n.  
- The name of the command needs to be understandable, check Naming convention 
  - I changed the name of the command to `Translator`.
- Do I need to register for a key before “history” and does that make any sense?
  - No. You don't need to register for a key to use History. It isn't related.
- We do already have Google Translate so it makes sense to make the Papago translater and focus on that?
  - Translation services are not accurate. so, It usually need to compare multiple services for better understanding.

pernielsentikaer commented on extensions/tt/package.json:
```diff
@@ -0,0 +1,929 @@
+  "commands": [
+    {
+      "name": "index",
+      "title": "Transator",
```
```suggestion
      "title": "Translate Text",
```

pernielsentikaer commented on extensions/tt/package.json:
```diff
@@ -0,0 +1,929 @@
+    {
+      "name": "index",
+      "title": "Transator",
+      "subtitle": "Translate texts",
```
```suggestion
      "subtitle": "Papago",
```

pernielsentikaer commented on extensions/tt/package.json:
```diff
@@ -0,0 +1,929 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "tt",
+  "title": "TT",
```
```suggestion
  "title": "Papago Translator",
```

pernielsentikaer commented on extensions/tt/package.json:
```diff
@@ -0,0 +1,929 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "tt",
```
```suggestion
  "name": "papago-translator",
```

pernielsentikaer commented on extensions/tt/src/component/T.tsx:
```diff
@@ -0,0 +1,73 @@
+
+  return (
+    <List
+      navigationTitle={L.TT}
```
```suggestion
```

pernielsentikaer reviewed (changes requested):
Hi @deptno 👋 

Thanks for the update 💪 

I have some suggestions and comments for you to address:

- I'm not sure how the `history` works, tried a lot of searches but seems like nothing ends in the history?
- Are you in Slack, would love to have an API key for 15 minutes to test it fully out.

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

deptno commented:
@pernielsentikaer 

Thank you for your feedbacks.

1. Purpose of the extension is to serve multiple translated texts from multiple providers(Google, Papago, etc) at once.

So, "Papago" can't be name of the extension.

2. History doesn't work automatically, It is triggered by user action(`cmd` + `enter` or ActionPanel)
`History` title is confusing, I'll change `History` to something like `Starred`?

3. I tried to join your slack channel but I failed to join with message below.
```
deptno@gmail.com does not have an account for this workspace.
```
By the way, Use my key below.
![image](https://user-images.githubusercontent.com/1223020/156967292-141b5d7c-76cf-4fe8-af95-f90f845afeca.png)
```
X-Naver-Client-Id: 0hRo0SzNWoxFPsaBn9aD
X-Naver-Client-Secret: IRgliRw0Tv
```

Notify me when you have done tested feature. I'll revoke my API key.

pernielsentikaer reviewed (changes requested):
Hi @deptno 👋 

1) Makes sense 😄 
2) Ah, seems like a section title caused the confusion

I have some suggestions for you:

1) I would move the credentials to preferences instead, would fit how the rest of Raycast is working.
2) After adding the key, then it seems like it's not happy fetching the result 😄 

![fG0lgifaD40tDszW](https://user-images.githubusercontent.com/38033107/156971967-6cd449a5-6900-4c8f-80c7-9a21460bc47b.png)

3) Icon is missing in the ActionPanel here:

![ZhqUeeST6Xz5XLLf](https://user-images.githubusercontent.com/38033107/156972149-9238475c-0ea1-43ca-aae4-b443e7df214f.png)

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

deptno commented:
@pernielsentikaer

Thank you for feedbacks!

I agree that section title caused the confusion. so I removed `History` title from the section name.

I have question. Can you tell me your preferences?
- From 
- To

I converted PR status to draft. I will mention you again soon.

pernielsentikaer commented:
> I have question. Can you tell me your preferences values?

I selected Danish to English 😄

pernielsentikaer commented:
Hi @deptno 👋 

Do you have an update here?

deptno commented:
Hi @pernielsentikaer 

I'm not able to work on this now.
I will Pull request again later.
